### PR TITLE
Editable: Rename the Editable component, RichText component for clarity

### PR DESCRIPTION
**Pull request by youknowriad**
refs #4776

This PR renames the `Editable` component `RichText` to clarify that it's meant to be used for Rich Content exclusively and not as a replacement to inputs/textareas.

We still export an `Editable` component for backwards compatibility with a warning asking people to use `RichText` instead.

Next step will be to implement a `PlainText` component and use it in one/two blocks. The blocks where we're currently using `TextareaAutosize`.

**Testing instructions**

 - Open the editor
 - Insert different content blocks (paragraph, list, quotes, image with captions...)
 - Check that they still work and look as expected.

### CODE REVIEW

File: blocks/rich-text/editable.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Internal dependencies
+ */
+import RichText from './';
+
+class Editable extends RichText {
+	constructor() {
+		super( ...arguments );
+		// eslint-disable-next-line no-console
+		console.warn( 'Editable is deprecated use wp.blocks.RichText instead.' );
```
Comment (iseulde) [inline blocks/rich-text/editable.js]: Minor: this sentence could use a period. :)

**Review by ellatrix — APPROVED** [review]
Catching some "editable" mentions in `README.md`, but nothing that doesn't make sense as it's still an editable field. :)

**Comment by aduth** [comment]
I'd be in favor of this change.

File: blocks/rich-text/editable.js (hunk 1 above)
Comment (youknowriad) [reply blocks/rich-text/editable.js]: It already has a period?

**Comment by gziolo** [comment]
This change is important enough to make sure is noticed, cc @mtias @karmatosed.

**Comment by iseulde** [comment]
I updated the "Editable" label to "Rich Text".

File: blocks/rich-text/editable.js (hunk 1 above)
Comment (iseulde) [reply blocks/rich-text/editable.js]: Yes, I did. :) Sorry, should have been more specific :)

**Comment by zgordon** [comment]
Is Editable still available as a name for backwards compatibility? Or do folks need to switch to RichText right away? Thanks!!!

**Comment by youknowriad** [comment]
> Is Editable still available as a name for backwards compatibility? Or do folks need to switch to RichText right away? Thanks!!!

It's available and will be removed in like two or three releases ;)

**Comment by zgordon** [comment]
👌

**Pull request merged**